### PR TITLE
valgrind: disable mpicc and make Linux only

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This disables `mpicc` support in valgrind, which seems to cause build failures on host systems with `mpich` or OpenMPI installed. If we ever get a request to add support for `mpicc` to `valgrind`, we'd probably have to add `mpich` or `open-mpi` as a dependency.  Additionally the configure argument `--build=amd64-darwin` should only be used on macOS.

Moved to homebrew-core from https://github.com/Homebrew/linuxbrew-core/pull/21870.

### CODE REVIEW

File: Formula/valgrind.rb [hunk 1]
```diff
@@ -36,8 +36,11 @@ def install
       --disable-dependency-tracking
       --prefix=#{prefix}
       --enable-only64bit
-      --build=amd64-darwin
+      --without-mpicc
     ]
+
+    args << "--build=amd64-darwin" if OS.mac?
```
Comment (SeekingMeaning) [inline Formula/valgrind.rb:42]:
```suggestion
    on_macos do
      args << "--build=amd64-darwin"
    end
```

**Comment by SMillerDev** [comment]
This formula should probably just be linux only because it doesn't run on any of our supported macOS versions

**Comment by carlocab** [comment]
`depends_on maximum_macos: :sierra` would work too, though I can see the advantages of it being Linux-only.

There also seems to be a version of valgrind that works at least until Catalina: https://github.com/LouisBrunner/valgrind-macos

**Comment by danielnachun** [comment]
Since there is a tap that provides a working version of `valgrind` for all versions of macOS, I'm inclined to make this Linux only.  Hopefully the patches in that fork of `valgrind` can be integrated into the official release so that it will once again work in macOS, but that is an issue for upstream to deal with.

**Comment by danielnachun** [comment]
Actually if we make this Linux only, should I close this pull request?  Does homebrew-core allow Linux only formulae?

**Comment by SeekingMeaning** [comment]
> Does homebrew-core allow Linux only formulae?

Yep, for example: https://github.com/Homebrew/homebrew-core/pull/65300

**Comment by danielnachun** [comment]
Thanks for the revisions, I'll try removing skip_clean on my Linux install and see if it breaks anything.  Everything else looks pretty straightforward.

**Comment by SeekingMeaning** [comment]
Also add `Formula/valgrind.rb` to `.github/workflows/tests-linux.yml`

**Comment by iMichka** [comment]
I agree that this can be made linux-only. Though we also try to disable valgrind support when possible in all other formulae.

**Review by SeekingMeaning — APPROVED** [review]

**Review by iMichka — APPROVED** [review]

**Pull request merged**